wp-content/plugins/wporg-learn/inc/capabilities.php

```diff
@@ -0,0 +1,237 @@
+<?php
+
+namespace WPOrg_Learn\Capabilities;
+
+defined( 'WPINC' ) || die();
+
+/**
+ * Actions and filters.
+ */
+add_filter( 'user_has_cap', __NAMESPACE__ . '\set_post_type_caps' );
+add_filter( 'user_has_cap', __NAMESPACE__ . '\set_caps_for_internal_notes' );
+add_filter( 'map_meta_cap', __NAMESPACE__ . '\map_meta_caps', 20, 4 ); // Needs to fire after meta caps in wporg-internal-notes.
+add_action( 'init', __NAMESPACE__ . '\add_or_update_lesson_plan_editor_role' );
+add_action( 'init', __NAMESPACE__ . '\add_or_update_workshop_reviewer_role' );
+
+/**
+ * Assign custom post type caps to roles based on their equivalents for the 'post' post type.
+ *
+ * Example: If a user has the `edit_others_posts` cap (from Editor role), this will also give them
+ * the equivalent `edit_others_lesson_plans` and `edit_others_workshops` caps.
+ *
+ * @param bool[] $user_caps A list of primitive caps (keys) and whether user has them (boolean values).
+ *
+ * @return array
+ */
+function set_post_type_caps( $user_caps ) {
+	$capability_types = array(
+		array( 'lesson_plan', 'lesson_plans' ),
+		array( 'workshop', 'workshops' ),
+	);
+
+	foreach ( $capability_types as $capability_type ) {
+		// Generate the caps for a capability type.
+		$cap_args = array(
+			'capability_type' => $capability_type,
+			'capabilities'    => array(),
+			'map_meta_cap'    => true,
+		);
+		$cap_map = (array) get_post_type_capabilities( (object) $cap_args );
+
+		foreach ( $user_caps as $cap => $bool ) {
+			if ( $bool && isset( $cap_map[ $cap ] ) ) {
+				$user_caps[ $cap_map[ $cap ] ] = true;
```

Comment:
```suggestion
		foreach ( $user_caps as $cap_slug => $granted ) {
			if ( $granted && isset( $cap_map[ $cap_slug ] ) ) {
				$user_caps[ $cap_map[ $cap_slug ] ] = true;
```

`$cap` feels ambiguous to me, and `$bool` describes the data type, rather than what the variable means.